### PR TITLE
Habitat plan for Windows

### PR DESCRIPTION
**Pull request by miah**
A Windows habitat plan.ps1 for InSpec!

Heavily cribbed from @robbkidd :1st_place_medal: work on [Infra](https://github.com/chef/chef/pull/8900).

### CODE REVIEW

**Comment by miah** [comment]
I'm flipping the bit on this, lets get it merged. 

There is still work to do with the testing pipeline, and there are probably lots of bugs to be discovered. Definitely don't consider this supported at this point =)

File: test/artifact/inspec_detect_test.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require "minitest/autorun"
+
+class TestArtifactDetect < Minitest::Test
+  def test_detect
+    out, err = capture_subprocess_io {
+      assert system("inspec detect --no-color")
+    }
+
+    assert_match %r%Platform Details%, out
```
Comment (zenspider) [inline test/artifact/inspec_detect_test.rb]: won't chefstyle get pissy about using `%r` when it doesn't have `/` in the pattern?

Huh. apparently not. But it does  blow up on the use of curlies on the assertion above. I agree with the curlies. so you might want to disable that check. I think that's the only blocker to getting the tests running.

**Review by zenspider — COMMENTED** [review]
minor feedback about BK being red on chefstyle

File: inspec-bin/Rakefile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Bundler::GemHelper.install_tasks(name: "inspec-bin")
+
+desc "force install the inspec-bin gem"
+task "install:force" do
+  sh "gem build -V inspec-bin.gemspec"
+  built_gem_path = Dir["inspec-bin-*.gem"].sort_by { |f| File.mtime(f) }.last
```
Comment (zenspider) [inline inspec-bin/Rakefile:6]:
```ruby
[...].sort_by { |f| File.mtime(f) }.last
```
vs
```ruby
[...].max_by { |f| File.mtime(f) }
```

File: test/artifact/Rakefile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require "rake/testtask"
+
+Rake::TestTask.new(:test) do |t|
+  t.test_files = FileList["*_test.rb"].sort
```
Comment (zenspider) [inline test/artifact/Rakefile:4]: thank you for sorting this... file systems are strange beasts.

**Review by clintoncwolfe — COMMENTED** [review]
I'm OK with approving everything except the plan PowerShell itself - I just don't know enough PowerShell, let alone Habitat plan idioms, to be a competent reviewer here. Could we get someone who works on windows plans frequently to review this and give feedback?

**Pull request merged**